stonesoup/models/transition/linear.py

```diff
@@ -30,27 +30,27 @@ def ndim_state(self):
         return self.matrix().shape[0]
 
 
-class CombinedLinearGaussianTransitionModel(LinearModel, _CombinedGaussianTransitionModel):
-    r"""Combine multiple models into a single model by stacking them.
+# class CombinedLinearGaussianTransitionModel(LinearModel, _CombinedGaussianTransitionModel):
+#     r"""Combine multiple models into a single model by stacking them.
 
-    The assumption is that all models are Linear and Gaussian.
-    Time Variant, and Time Invariant models can be combined together.
-    If any of the models are time variant the keyword argument "time_interval"
-    must be supplied to all methods
-    """
+#     The assumption is that all models are Linear and Gaussian.
+#     Time Variant, and Time Invariant models can be combined together.
+#     If any of the models are time variant the keyword argument "time_interval"
+#     must be supplied to all methods
+#     """
 
-    def matrix(self, **kwargs):
-        """Model matrix :math:`F`
+#     def matrix(self, **kwargs):
+#         """Model matrix :math:`F`
 
-        Returns
-        -------
-        : :class:`numpy.ndarray` of shape\
-        (:py:attr:`~ndim_state`, :py:attr:`~ndim_state`)
-        """
+#         Returns
+#         -------
+#         : :class:`numpy.ndarray` of shape\
+#         (:py:attr:`~ndim_state`, :py:attr:`~ndim_state`)
+#         """
 
-        transition_matrices = [
-            model.matrix(**kwargs) for model in self.model_list]
-        return block_diag(*transition_matrices)
+#         transition_matrices = [
+#             model.matrix(**kwargs) for model in self.model_list]
+#         return block_diag(*transition_matrices)
```

Comment:
If the entire class is commented-out, it should probably be deleted. Though I think having the 'linear' combined model is still useful. Applying the combined transition via a single matrix multiplication will be quicker than the model-by-model logic of the combined parent class.

Reply:
Yes, it looks like I forgot to remove that.
To handle time varying models, the total matrix has to be rebuilt at each call. Also because the total matrix is block diagonal, the full matrix multiply involves a lot of multiply by zero operations. It may make sense to do the smaller matrix multiplies and stitch together the result. It would probably be useful to benchmark the results to see if there is a significant performance hit.

Reply:
@oharrald-Dstl @sdhiscocks  I've done some benchmarking and indeed the CombinedLinearGaussianTransitionModel is faster than the general implementation. I will add it back in, but utilize the new class hierarchy.